libff/algebra/curves/alt_bn128/alt_bn128_g1.cpp

```diff
@@ -405,31 +405,51 @@ alt_bn128_G1 alt_bn128_G1::random_element()
     return (scalar_field::random_element().as_bigint()) * G1_one;
 }
 
-std::ostream& operator<<(std::ostream &out, const alt_bn128_G1 &g)
+void alt_bn128_G1::write_uncompressed(std::ostream &out) const
 {
-    alt_bn128_G1 copy(g);
+    alt_bn128_G1 copy(*this);
     copy.to_affine_coordinates();
 
     out << (copy.is_zero() ? 1 : 0) << OUTPUT_SEPARATOR;
-#ifdef NO_PT_COMPRESSION
     out << copy.X << OUTPUT_SEPARATOR << copy.Y;
-#else
+}
+
+void alt_bn128_G1::write_compressed(std::ostream &out) const
+{
+    alt_bn128_G1 copy(*this);
+    copy.to_affine_coordinates();
+
+    out << (copy.is_zero() ? 1 : 0) << OUTPUT_SEPARATOR;
     /* storing LSB of Y */
     out << copy.X << OUTPUT_SEPARATOR << (copy.Y.as_bigint().data[0] & 1);
-#endif
-
-    return out;
 }
 
-std::istream& operator>>(std::istream &in, alt_bn128_G1 &g)
+void alt_bn128_G1::read_uncompressed(std::istream &in, alt_bn128_G1 &g)
 {
     char is_zero;
     alt_bn128_Fq tX, tY;
 
-#ifdef NO_PT_COMPRESSION
     in >> is_zero >> tX >> tY;
     is_zero -= '0';
-#else
+
+    // using Jacobian coordinates
+    if (!is_zero)
+    {
+        g.X = tX;
+        g.Y = tY;
+        g.Z = alt_bn128_Fq::one();
```

Comment:
Why don't we check that the point is on the curve first?
Seems to me that the deserialization logic of a point in the affine space should be:
1. Recover `x`, `y`
2. Check that the tuple `(x, y)` is one the curve, i.e. satisfies the curve equation
3. Either do the "safe subgroup membership test" or mult by cofactor to make sure we have a point in G1 and not a point in the E/Fq group (which may be of lower order).

Do I miss something (but this code initially originates from the library, and has just been moved around in this PR - so this question applies to the current state of the library, as in master, for the deserialization of uncompressed points)

Reply:
Note, it seems to me that, the same assumption is made in the deserialization of compressed points, since I can read here that the `sqrt` function does not terminate if the element is a quadratic non-residue in the field: https://github.com/scipr-lab/libff/blob/master/libff/algebra/fields/fp.hpp#L108

We can "skip" the bullet `3.` above here since cofactor = 1, but why skipping bullet `2.`?